pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -83,18 +92,19 @@ func updateReplicaSetStatus(logger klog.Logger, c appsclient.ReplicaSetInterface
 	return nil, updateErr
 }
 
-func calculateStatus(rs *apps.ReplicaSet, filteredPods []*v1.Pod, manageReplicasErr error) apps.ReplicaSetStatus {
+func calculateStatus(rs *apps.ReplicaSet, activePods []*v1.Pod, terminatingPods []*v1.Pod, manageReplicasErr error) apps.ReplicaSetStatus {
 	newStatus := rs.Status
 	// Count the number of pods that have labels matching the labels of the pod
 	// template of the replica set, the matching pods may have more
 	// labels than are in the template. Because the label of podTemplateSpec is
 	// a superset of the selector of the replica set, so the possible
-	// matching pods must be part of the filteredPods.
+	// matching pods must be part of the activePods.
 	fullyLabeledReplicasCount := 0
 	readyReplicasCount := 0
 	availableReplicasCount := 0
+	var terminatingReplicasCount *int32
```

Comment:
nit: move this down to just above the `if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentPodReplacementPolicy) {` block to make it clearer nothing in the loop over `activePods` cares about this